src/main/java/com/amazon/janusgraph/graph/AbstractGraph.java

```diff
@@ -0,0 +1,30 @@
+package com.amazon.janusgraph.graph;
+
+import com.amazon.janusgraph.triple.Triple;
+import org.janusgraph.core.JanusGraph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URL;
+import java.util.Set;
+import java.util.concurrent.*;
+
+
+/**
+ * Created by addisonslabaugh on 6/21/17.
+ */
+public interface AbstractGraph {
```

Comment:
The goal of a minimal reproduction of a failure is to be as succinct as possible. Ideally, all your code would fit in ScenarioTests.java. However, your example is complicated, so I want to get it in the right place and reproducing first, before making it lean and mean.